runtime/tracing/tracing-core/common/test/aws/smithy/kotlin/runtime/tracing/TraceSpanTest.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.tracing
+
+import io.kotest.matchers.string.shouldContain
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNull
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class TraceSpanTest {
```

Comment:
**Nit**: This is really testing the `CoroutineContext` utils, not the `TraceSpan` interface. We should probably rename the file/class.